java/spark/src/main/java/com/lancedb/lance/spark/internal/LanceDatasetAdapter.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.lancedb.lance.spark.internal;
+
+import com.lancedb.lance.Dataset;
+import com.lancedb.lance.DatasetFragment;
+import com.lancedb.lance.Fragment;
+import com.lancedb.lance.FragmentMetadata;
+import com.lancedb.lance.FragmentOperation;
+import com.lancedb.lance.WriteParams;
+import com.lancedb.lance.spark.LanceConfig;
+import com.lancedb.lance.spark.read.LanceInputPartition;
+import com.lancedb.lance.spark.utils.Optional;
+import com.lancedb.lance.spark.write.InternalRowWriterArrowReader;
+import org.apache.arrow.c.ArrowArrayStream;
+import org.apache.arrow.c.Data;
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.memory.RootAllocator;
+import org.apache.arrow.vector.ipc.ArrowReader;
+import org.apache.spark.sql.types.StructType;
+import org.apache.spark.sql.util.ArrowUtils;
+
+import java.time.ZoneId;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class LanceDatasetAdapter {
+  private static final BufferAllocator allocator = new RootAllocator(
+      RootAllocator.configBuilder().from(RootAllocator.defaultConfig())
+          .maxAllocation(4 * 1024 * 1024).build());
+
+  public static Optional<StructType> getSchema(LanceConfig config) {
+    return getSchema(config.getTablePath());
+  }
+
+  public static Optional<StructType> getSchema(String tablePath) {
```

Comment:
This does not look right. So we have `table` concept and `Dataset` concept here.  We need to be consistent. 

Also, can we just call this class `LanceDataset`?

Reply:
this class is for having a central place to put allocator and put all methods requires allocator in the same space

The LanceTable has been renamed to LanceDataset which allows user to read or write a single lance dataset